src/js/index.js

```diff
@@ -21,15 +21,73 @@ const allowDifficultySelect = shouldAllow => sortingInput.querySelectorAll('.dif
 		node.disabled = !shouldAllow;
 	});
 
+const parameters = {
+	tags: {
+		default: '',
+		getValue: () => selectr.getValue().join(' '),
+		setValue: value => selectr.setValue(value.split(' '))
+	},
+	difficulty: {
+		default: 'all',
+		getValue: () => filterInput.value,
+		setValue: value => {
+			filterInput.value = ['easy', 'medium', 'hard'].includes(value) ?
+				value :
+				'all';
+		}
+	},
+	sort: {
+		default: 'alphabetical',
+		getValue: () => sortParams.sort.toLowerCase(),
+		setValue: value => {
+			sortParams.sort = ['alphabetical', 'difficulty'].includes(value) ?
+				value.toUpperCase() :
+				'ALPHABETICAL';
+		}
+	},
+	order: {
+		default: 'asc',
+		getValue: () => sortParams.order.toLowerCase(),
+		setValue: value => {
+			sortParams.order = ['asc', 'desc'].includes(value) ?
+				value.toUpperCase() :
+				'ASC';
+		}
+	}
+};
+
+const sortParams = {
+	sort: null,
+	order: null
+};
+
 let search;
 let selectr;
 
+function updateUrl() {
+	const newSearch = new URLSearchParams(window.location.search);
+	for (const [paramName, paramObj] of Object.entries(parameters)) {
+		const paramValue = paramObj.getValue();
+		if (['', paramObj.default].includes(paramValue)) {
```

Comment:
```suggestion
		if (paramValue.length === 0 || paramValue === paramObj.default) {
```

Makes the code more readable and clear, in my opinion.